### PR TITLE
Code: check whether the LSP binary is in PATH

### PR DESCRIPTION
**Pull request by lnicola**
I'm not really sure about the TS changes. I just made a couple of functions async and it seems to work.

### CODE REVIEW

File: editors/code/src/extension.ts [hunk 1]
```diff
@@ -89,13 +89,9 @@ export function activate(context: vscode.ExtensionContext) {
     }
 
     // Notifications are events triggered by the language server
-    const allNotifications: Iterable<
```
Comment (lnicola) [inline editors/code/src/extension.ts]: The original indentation here is arguably bad, and we've kept running into problems with this one and the other. My version of `prettier` picks a different indentation than the one on CI.

**Comment by matklad** [comment]
bors r+

Thanks!

**Comment by edwin0cheng** [comment]
@lnicola `lookpath` seem to be not working in Windows. :(

**Comment by lnicola** [comment]
Ugh, what, why? :-(.

**Comment by edwin0cheng** [comment]
https://github.com/otiai10/lookpath/issues/4

**Comment by matklad** [comment]
Perhaps a better approach is just to try to spawn `rust-analyzer --version`?

**Pull request merged**